src/serial.rs

```diff
@@ -191,8 +225,8 @@ impl<W: Write> Serial<W> {
         (self.modem_control & MCR_LOOP_BIT) != 0
     }
 
-    fn trigger_interrupt(&mut self) -> Result<()> {
-        self.interrupt_evt.write(1)
+    fn trigger_interrupt(&mut self) -> std::result::Result<(), T::E> {
```

Comment:
Since we are using this std::result::Result quite a lot, how about we define a `Result` wrapper type?